pkg/noderesourcetopology/filter.go

```diff
@@ -71,22 +70,15 @@ func resMatchNUMANodes(numaNodes NUMANodeList, resources v1.ResourceList, qos v1
 			// if the resource can be found at the node itself, because there are resources which are not NUMA aligned
 			// or not supported by the topology exporter - if resource was not found at both checks - skip (don't set it as available NUMA node).
 			// if the un-found resource has 0 quantity probably this numa node can be considered.
-			if !ok && !resourceFoundOnNode(resource, quantity, nodeInfo) && quantity.Cmp(zeroQuantity) != 0 {
+			if !ok && !resourceFoundOnNode(resource, quantity, nodeInfo) && !quantity.IsZero() {
 				continue
 			}
-			// Check for the following:
-			// 1. set numa node as possible node if resource is memory or Hugepages
-			// 2. set numa node as possible node if resource is cpu and it's not guaranteed QoS, since cpu will flow
-			// 3. set numa node as possible node if zero quantity for non existing resource was requested
-			// 4. otherwise check amount of resources
-			if resource == v1.ResourceMemory ||
-				strings.HasPrefix(string(resource), v1.ResourceHugePagesPrefix) ||
-				resource == v1.ResourceCPU && qos != v1.PodQOSGuaranteed ||
-				quantity.Cmp(zeroQuantity) == 0 ||
-				numaQuantity.Cmp(quantity) >= 0 {
-				// possible to align resources on NUMA node
-				resourceBitmask.Add(numaNode.NUMAID)
+
+			if !isNUMANodeSuitable(qos, resource, quantity, numaQuantity) {
+				continue
 			}
+
+			resourceBitmask.Add(numaNode.NUMAID)
 		}
 		bitmask.And(resourceBitmask)
 		if bitmask.IsEmpty() {
```

Comment:
IIUC this function returns true if there is a matching numa to fit the resources. 
this code returns once the numa matching is found for one source (the first), don't we want to loop over the rest?

Reply:
the function returns true if we need to discard the set of NUMA zones (pertaining to a compute node), false if at least a NUMA zone among all the provided ones can accomodate the workload, as defined as by its resources footprint (`resources` argument)

Reply:
filed https://github.com/kubernetes-sigs/scheduler-plugins/issues/351 - I'll evaluate if sending one or two PRs (testing + refactoring vs all in one)